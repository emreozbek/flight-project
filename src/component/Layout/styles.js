export const styles = theme => ({
  background: {
    display: "block",
    content: "",
    width: "100%",
    height: theme.spacing.unit * 30,
    backgroundColor: theme.palette.secondary.main,
    position: "absolute",
    zIndex: -1
  },
  wrapper: {
    padding: `${theme.spacing.unit * 8}px ${theme.spacing.unit * 2}px 0 ${theme
      .spacing.unit * 2}px`,
    maxWidth: 720 + theme.spacing.unit * 4,
    marginLeft: "auto",
    marginRight: "auto"
  }
});
