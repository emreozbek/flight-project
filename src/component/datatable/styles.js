export const styles = theme => ({
  table: {
    minWidth: 720
  },
  tableWrapper: {
    overflowX: "auto"
  },
  searchBox: {
    paddingTop: theme.spacing.unit * 2,
    paddingLeft: theme.spacing.unit * 2,
    paddingRight: theme.spacing.unit * 2,
    paddingBottom: theme.spacing.unit * 2
  }
});
