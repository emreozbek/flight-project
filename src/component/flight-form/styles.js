export const styles = theme => ({
  padding: {
    padding: theme.spacing.unit * 2
  },
  marginLeft: {
    marginLeft: theme.spacing.unit * 2
  }
});
