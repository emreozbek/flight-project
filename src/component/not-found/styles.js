export const styles = theme => ({
  padding: {
    padding: theme.spacing.unit * 2
  }
});
