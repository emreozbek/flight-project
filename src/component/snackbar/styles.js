export const styles = theme => ({
  close: {
    padding: theme.spacing.unit / 2
  },
  error: {
    backgroundColor: theme.palette.error.main
  }
});
