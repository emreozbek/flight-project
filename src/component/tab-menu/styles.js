export const styles = theme => ({
  content: {
    position: "relative",
    minHeight: 256
  }
});
