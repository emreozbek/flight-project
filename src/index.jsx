import React from "react";
import { render } from "react-dom";
import { Provider } from "react-redux";
import { Router } from "react-router";
import App from "./component/App";
import { store, history } from "./store";

render(
  <Provider store={store}>
    <Router history={history}>
      <App />
    </Router>
  </Provider>,
  document.getElementById("app")
);
