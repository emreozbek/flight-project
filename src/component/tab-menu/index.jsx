import React, { Component, Fragment } from "react";
import PropTypes from "prop-types";
import { Route, Link } from "react-router-dom";
import { Tab, Tabs, withStyles } from "@material-ui/core";
import FlightList from "../flight-list";
import FlightForm from "../flight-form";
import { styles } from "./styles";
import { history } from "../../store";

export class TabMenu extends Component {
  state = { page: window.location.pathname };
  componentDidMount = () => {
    this.unListen = history.listen(location => {
      this.setState({ page: location.pathname });
    });
  };
  componentWillUnmount = () => {
    this.unListen();
  };
  render = () => {
    const { classes } = this.props;
    const { page } = this.state;
    return (
      <Fragment>
        <Tabs
          value={page}
          variant="fullWidth"
          onChange={(e, page) => this.setState({ page })}
          centered
        >
          <Tab label="Flights" component={Link} to="/" value="/" />
          <Tab label="Add FLight" component={Link} to="/add" value="/add" />
        </Tabs>
        <div className={classes.content}>
          <Route exact path="/" component={FlightList} />
          <Route path="/add" component={FlightForm} />
        </div>
      </Fragment>
    );
  };
}
TabMenu.propTypes = {
  classes: PropTypes.object.isRequired
};
export default withStyles(styles)(TabMenu);
