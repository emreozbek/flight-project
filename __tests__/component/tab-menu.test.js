import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { withStyles } from "@material-ui/core";
import { Provider } from "react-redux";
import { TabMenu } from "../../src/component/tab-menu";
import { styles } from "../../src/component/tab-menu/styles";
import { store } from "../../src/store";

const TabMenuStyled = withStyles(styles)(TabMenu);

describe("TabMenu", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <TabMenuStyled />
          </Router>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
