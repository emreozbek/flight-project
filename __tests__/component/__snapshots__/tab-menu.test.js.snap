// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabMenu renders correctly 1`] = `
Array [
  <div
    className="MuiTabs-root-2"
  >
    <div
      className="MuiTabs-flexContainer-3"
    >
      <div
        className="MuiTabs-scroller-5 MuiTabs-fixed-6"
        onScroll={[Function]}
        role="tablist"
        style={
          Object {
            "marginBottom": 0,
          }
        }
      >
        <div
          className="MuiTabs-flexContainer-3 MuiTabs-centered-4"
        >
          <a
            aria-selected={true}
            className="MuiButtonBase-root-23 MuiTab-root-11 MuiTab-textColorInherit-13 MuiTab-selected-16 MuiTab-fullWidth-18"
            href="/"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
          >
            <span
              className="MuiTab-wrapper-19"
            >
              <span
                className="MuiTab-labelContainer-20"
              >
                <span
                  className="MuiTab-label-21"
                >
                  Flights
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-88"
            />
          </a>
          <a
            aria-selected={false}
            className="MuiButtonBase-root-23 MuiTab-root-11 MuiTab-textColorInherit-13 MuiTab-fullWidth-18"
            href="/add"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex="0"
          >
            <span
              className="MuiTab-wrapper-19"
            >
              <span
                className="MuiTab-labelContainer-20"
              >
                <span
                  className="MuiTab-label-21"
                >
                  Add FLight
                </span>
              </span>
            </span>
            <span
              className="MuiTouchRipple-root-88"
            />
          </a>
        </div>
        <span
          className="MuiPrivateTabIndicator-root-26 MuiPrivateTabIndicator-colorSecondary-28 MuiTabs-indicator-10"
          style={
            Object {
              "left": 0,
              "width": 0,
            }
          }
        />
      </div>
    </div>
  </div>,
  <div
    className="TabMenu-content-1"
  >
    <div
      className="overlay"
    >
      <div
        className="loading-animation"
      >
        <div
          className="lds-grid"
        >
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
          <div />
        </div>
      </div>
    </div>
    <div
      className="DataTable-searchBox-31"
    >
      <div
        className="MuiInputBase-root-32 MuiInputBase-fullWidth-41 MuiInputBase-adornedEnd-37"
        onClick={[Function]}
      >
        <input
          className="MuiInputBase-input-42 MuiInputBase-inputAdornedEnd-48"
          defaultValue=""
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search in the list"
          type="text"
        />
        <div
          className="MuiInputAdornment-root-49 MuiInputAdornment-positionEnd-52"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root-54"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
            />
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="DataTable-tableWrapper-30"
    >
      <table
        aria-labelledby="tableTitle"
        className="MuiTable-root-63 DataTable-table-29"
      >
        <thead
          className="MuiTableHead-root-64"
        >
          <tr
            className="MuiTableRow-root-65 MuiTableRow-head-68"
          >
            <th
              aria-sort="ascending"
              className="MuiTableCell-root-70 MuiTableCell-head-71"
              scope="col"
            >
              <span
                className="MuiButtonBase-root-23 MuiTableSortLabel-root-82 MuiTableSortLabel-active-83"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                Flight Type
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-54 MuiTableSortLabel-icon-84 MuiTableSortLabel-iconDirectionAsc-86"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-70 MuiTableCell-head-71"
              scope="col"
            >
              <span
                className="MuiButtonBase-root-23 MuiTableSortLabel-root-82"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                Departure
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-54 MuiTableSortLabel-icon-84 MuiTableSortLabel-iconDirectionAsc-86"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-70 MuiTableCell-head-71"
              scope="col"
            >
              <span
                className="MuiButtonBase-root-23 MuiTableSortLabel-root-82"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                Departure Date
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-54 MuiTableSortLabel-icon-84 MuiTableSortLabel-iconDirectionAsc-86"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-70 MuiTableCell-head-71"
              scope="col"
            >
              <span
                className="MuiButtonBase-root-23 MuiTableSortLabel-root-82"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                Arrival
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-54 MuiTableSortLabel-icon-84 MuiTableSortLabel-iconDirectionAsc-86"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
            <th
              aria-sort={null}
              className="MuiTableCell-root-70 MuiTableCell-head-71"
              scope="col"
            >
              <span
                className="MuiButtonBase-root-23 MuiTableSortLabel-root-82"
                onBlur={[Function]}
                onClick={[Function]}
                onContextMenu={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role="button"
                tabIndex="0"
              >
                Arrival Date
                <svg
                  aria-hidden="true"
                  className="MuiSvgIcon-root-54 MuiTableSortLabel-icon-84 MuiTableSortLabel-iconDirectionAsc-86"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
                  />
                </svg>
              </span>
            </th>
          </tr>
        </thead>
        <tbody
          className="MuiTableBody-root-87"
        />
      </table>
    </div>
  </div>,
]
`;
